pymc_marketing/mmm/components/adstock.py

```diff
@@ -0,0 +1,175 @@
+
+Each of these transformations is a subclass of
+:class:`pymc_marketing.mmm.components.adstock.AdstockTransformation`
+and defines a function that takes a time series and returns the adstocked
+version of it. The parameters of the function are the parameters
+of the adstock transformation.
+
+Examples
+--------
+Create a new adstock transformation:
+
+.. code-block:: python
+
+    class MyAdstock(AdstockTransformation):
+        def function(self, x, alpha):
+            return x * alpha
+
+        default_priors = {"alpha": {"dist": "HalfNormal", "kwargs": {"sigma": 1}}}
+
+"""
+
+import warnings
+
+from pymc_marketing.mmm.components.base import Transformation
+from pymc_marketing.mmm.transformers import (
+    ConvMode,
+    WeibullType,
+    delayed_adstock,
+    geometric_adstock,
+    weibull_adstock,
+    ConvMode
+)
+
+
+class AdstockTransformation(Transformation):
+    """Subclass for all adstock functions.
+
+    In order to use a custom saturation function, inherit from this class and define:
+
+    - `function`: a function that takes x to adstock x
+    - `default_priors`: dictionary with priors for every parameter in function
+
+    Consider the predefined subclasses as examples.
+
+    """
+
+    prefix: str = "adstock"
+
+    def __init__(
+        self,
+        l_max: int = 10,
+        normalize: bool = False,
+        mode: ConvMode = ConvMode.After,
+        priors: dict | None = None,
+        prefix: str | None = None,
+    ) -> None:
+        self.l_max = l_max
+        self.normalize = normalize
+        self.mode = mode
+
+        super().__init__(priors=priors, prefix=prefix)
+
+
+class GeometricAdstock(AdstockTransformation):
+    """Wrapper around geometric adstock function.
+
+    For more information, see :func:`pymc_marketing.mmm.transformers.geometric_adstock`.
+
+    """
+
+    def function(self, x, alpha):
+        return geometric_adstock(
+            x, alpha=alpha, l_max=self.l_max, normalize=self.normalize, mode=self.mode
+        )
+
+    default_priors = {"alpha": {"dist": "HalfNormal", "kwargs": {"sigma": 1}}}
+
+
+class DelayedAdstock(AdstockTransformation):
+    """Wrapper around delayed adstock function.
+
+    For more information, see :func:`pymc_marketing.mmm.transformers.delayed_adstock`.
+
+    """
+
+    def function(self, x, alpha, theta):
+        return delayed_adstock(
+            x,
+            alpha=alpha,
+            theta=theta,
+            l_max=self.l_max,
+            normalize=self.normalize,
+            mode=self.mode,
+        )
+
+    default_priors = {
+        "alpha": {"dist": "HalfNormal", "kwargs": {"sigma": 1}},
+        "theta": {"dist": "HalfNormal", "kwargs": {"sigma": 1}},
+    }
+
+
+class WeibullAdstock(AdstockTransformation):
+    """Wrapper around weibull adstock function.
+
+    For more information, see :func:`pymc_marketing.mmm.transformers.weibull_adstock`.
+
+    """
+
+    def __init__(
+        self,
+        l_max: int = 10,
```

Comment:
shall we remove the default value of `10`? The adstock functions above do not have a default value.

Reply:
Was this the default from DelayedSaturatedMMM?

Reply:
I think it is `12` (see https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/mmm/transformers.py#L157) . However, I was thinking about removing the default to ensure the users add them (and are conscious about their choice) .WDYT?

Reply:
Remove the default of 10.

Reply:
This default is still there, did we agree to remove it?

Reply:
I removed it in this PR: https://github.com/cetagostini-wise/pymc-marketing/pull/5